THRUST_CUB_NS_POSTFIX redefined when backend is THRUST_DEVICE_SYSTEM_OMP

This is an update of https://developer.nvidia.com/nvidia_bug/2065814
It is still happening as of commit `3af276e` Wed Mar 20.


Simply including `thrust/device_vector` in a program compiled with `-DTHRUST_DEVICE_SYSTEM=THRUST_DEVICE_SYSTEM_OMP` causes compile warnings to spew to output.

Compiling this program:

```c++
#include <thrust/device_vector.h>

__host__
thrust::device_vector<float> test() {
  thrust::device_vector<float> v;
  return v;
}
```

emits many similar messages to 

```
In file included from /home/hughw/nvidia-bug/omp-build/thrust/thrust/system/cuda/detail/execution_policy.h:33,
                 from /home/hughw/nvidia-bug/omp-build/thrust/thrust/system/cuda/detail/util.h:33,
                 from /home/hughw/nvidia-bug/omp-build/thrust/thrust/system/cuda/detail/terminate.h:31,
                 from /home/hughw/nvidia-bug/omp-build/thrust/thrust/detail/allocator/temporary_allocator.inl:24,
                 from /home/hughw/nvidia-bug/omp-build/thrust/thrust/detail/allocator/temporary_allocator.h:84,
                 from /home/hughw/nvidia-bug/omp-build/thrust/thrust/detail/temporary_array.h:40,
                 from /home/hughw/nvidia-bug/omp-build/thrust/thrust/detail/overlapped_copy.h:23,
                 from /home/hughw/nvidia-bug/omp-build/thrust/thrust/detail/vector_base.inl:24,
                 from /home/hughw/nvidia-bug/omp-build/thrust/thrust/detail/vector_base.h:586,
                 from /home/hughw/nvidia-bug/omp-build/thrust/thrust/device_vector.h:25,
                 from test.cu:1:
/home/hughw/nvidia-bug/omp-build/thrust/thrust/system/cuda/config.h:73: warning: "THRUST_CUB_NS_PREFIX" redefined
 #define THRUST_CUB_NS_PREFIX namespace thrust {   namespace cuda_cub {
```

Please see attached repro code.

[repro.tar.gz](https://github.com/thrust/thrust/files/3004126/repro.tar.gz)

--- a/thrust/detail/allocator/temporary_allocator.inl
+++ b/thrust/detail/allocator/temporary_allocator.inl
@@ -20,7 +20,7 @@
 #include <thrust/system/detail/bad_alloc.h>
 #include <cassert>
 
-#ifdef __CUDACC__
+#if defined(__CUDA_ARCH__) && THRUST_DEVICE_SYSTEM == THRUST_DEVICE_SYSTEM_CUDA
 #include <thrust/system/cuda/detail/terminate.h>
 #endif
 
@@ -47,7 +47,7 @@ __host__ __device__
 
 #if !defined(__CUDA_ARCH__)
     throw thrust::system::detail::bad_alloc("temporary_buffer::allocate: get_temporary_buffer failed");
-#else
+#elif THRUST_DEVICE_SYSTEM == THRUST_DEVICE_SYSTEM_CUDA
     thrust::system::cuda::detail::terminate_with_message("temporary_buffer::allocate: get_temporary_buffer failed");
 #endif
   } // end if

